src/Common/test/Common.Security.Test/ConfigureCertificateOptionsTest.cs

```diff
@@ -2,52 +2,194 @@
         var options = new ConfigureCertificateOptions(configurationRoot);
         var opts = new CertificateOptions();
-        options.Configure(opts);
-        Assert.NotNull(opts.Certificate);
-        Assert.Equal(Microsoft.Extensions.Options.Options.DefaultName, opts.Name);
-        Assert.True(opts.Certificate.HasPrivateKey);
+
+        options.Configure(CertificateName, opts);
+
+        opts.Certificate.Should().NotBeNull();
+        CertificateName.Should().Be(opts.Name);
+        opts.Certificate!.HasPrivateKey.Should().BeTrue();
+    }
+
+    [Fact]
+    public void ConfigureCertificateOptions_ReadsPemFiles_CreatesCertificate()
+    {
+        const string certificateName = "test";
+        IConfigurationRoot configurationRoot = new ConfigurationBuilder().AddCertificate(certificateName, "instance.crt", "instance.key").Build();
+        var pemConfig = new ConfigureCertificateOptions(configurationRoot);
+
+        var opts = new CertificateOptions
+        {
+            Name = certificateName
+        };
+
+        pemConfig.Configure(certificateName, opts);
+
+        opts.Certificate.Should().NotBeNull();
+        certificateName.Should().BeEquivalentTo(opts.Name);
+        opts.Certificate!.HasPrivateKey.Should().BeTrue();
+    }
+
+    [Fact]
+    public async Task CertificateOptionsUpdateOnFileChange()
+    {
+        using var sandbox = new Sandbox();
+        string firstCertificateContent = await File.ReadAllTextAsync("instance.crt");
+        string firstPrivateKeyContent = await File.ReadAllTextAsync("instance.key");
+        var firstX509 = X509Certificate2.CreateFromPemFile("instance.crt", "instance.key");
+        string secondCertificateContent = await File.ReadAllTextAsync("instance2.crt");
+        string secondPrivateKeyContent = await File.ReadAllTextAsync("instance2.key");
+        var secondX509 = X509Certificate2.CreateFromPemFile("instance.crt", "instance.key");
+        string certificateFilePath = sandbox.CreateFile("cert", firstCertificateContent);
+        string privateKeyFilePath = sandbox.CreateFile("key", firstPrivateKeyContent);
+
+        IConfigurationRoot configuration = new ConfigurationBuilder().AddCertificate(CertificateName, certificateFilePath, privateKeyFilePath).Build();
+
+        ServiceProvider serviceProvider = new ServiceCollection().AddSingleton<IConfiguration>(configuration).ConfigureCertificateOptions(configuration, null)
+            .BuildServiceProvider();
+
+        var optionsMonitor = serviceProvider.GetRequiredService<IOptionsMonitor<CertificateOptions>>();
+
+        optionsMonitor.Get(CertificateName).Certificate.Should().BeEquivalentTo(firstX509);
+
+        await File.WriteAllTextAsync(certificateFilePath, secondCertificateContent);
+        await File.WriteAllTextAsync(privateKeyFilePath, secondPrivateKeyContent);
+        await Task.Delay(2000);
```

Comment:
There's no need to sleep when instead using `MemoryFileProvider.ReplaceFile/NotifyChanged`, see existing tests.

Reply:
Great suggestion, unfortunately this would require us to provide the `IFileProvider` via DI, which is probably not something Steeltoe should be responsible for